webui/react/src/pages/ModelVersionDetails/ModelVersionHeader.tsx

```diff
@@ -184,22 +140,57 @@ with det.import_from_path(path + "/code"):
     await copyToClipboard(referenceText);
   }, [referenceText]);
 
-  const menu: DropDownProps['menu'] = useMemo(() => {
-    const onItemClick: MenuProps['onClick'] = (e) => {
-      const action = actions.find((ac) => ac.key === e.key) as Action;
-      action.onClick();
-    };
-
-    const menuItems: MenuProps['items'] = actions.map((action) => ({
-      className: css.overflowAction,
-      danger: action.danger,
-      disabled: action.disabled,
-      key: action.key,
-      label: action.text,
-    }));
+  const menu = useMemo(() => {
+    const items: MenuItem[] = [
+      {
+        danger: false,
+        disabled: false,
+        key: MenuKey.DownloadModel,
+        label: MenuKey.DownloadModel,
+      },
+      {
+        danger: false,
+        disabled: false,
+        key: MenuKey.UseInNotebook,
+        label: MenuKey.UseInNotebook,
+      },
+      {
+        danger: false,
+        disabled: modelVersion.model.archived || !canModifyModelVersion({ modelVersion }),
+        key: MenuKey.EditModelVersionName,
+        label: MenuKey.EditModelVersionName,
+      },
+    ];
+    if (canDeleteModelVersion({ modelVersion })) {
+      items.push({
+        danger: true,
+        disabled: false,
+        key: MenuKey.DeregisterVersion,
+        label: MenuKey.DeregisterVersion,
+      });
```

Comment:
nit: I think the lines where `danger` or `disabled` are just `false` can be deleted, right?

Reply:
100%